zebra-chain/benches/block.rs

```diff
@@ -0,0 +1,28 @@
+use std::io::Cursor;
+
+use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion};
+
+use zebra_chain::{
+    block::Block,
+    serialization::{ZcashDeserialize, ZcashSerialize},
+};
+use zebra_test::vectors::BLOCK_TESTNET_141042_BYTES;
+
+fn block_serialization(c: &mut Criterion) {
+    // Biggest block from `zebra-test`.
+    let block_bytes: &[u8] = BLOCK_TESTNET_141042_BYTES.as_ref();
```

Comment:
@teor2345 do you think this largest block as the serialization benchmark is sufficient? I know our test vectors are not comprehensive in terms of the possible variety of outlier blocks (all transparent transactions, all shielded transactions of different types, just coinbase, etc) but that's more a test coverage/long tail issue than a benchmark issue.

Reply:
It depends on the purpose of the benchmark.

if it was security-related (for example, testing denial of service conditions), we'd want to try every different combination, probably using randomised testing.

For now I suggest we test:
- the largest public block
- the synthetic big blocks in our existing unit tests: https://github.com/ZcashFoundation/zebra/blob/main/zebra-chain/src/block/tests/generate.rs#L18

And then we can open a follow-up ticket for different transaction types. (We should probably test a big orchard transaction as part of the NU5 changes, I'll add it to our list in that ticket.)

Reply:
> And then we can open a follow-up ticket for different transaction types. (We should probably test a big orchard transaction as part of the NU5 changes, I'll add it to our list in that ticket.)

I created #2030 for sprout and sapling, and updated #1979 for orchard.

Reply:
Sorry I wasn't very clear here, let's test both:
- large_single_transaction_block
- large_multi_transaction_block (already implemented)

The performance of a large number of transactions vs a large number of inputs could be quite different.

Reply:
Added.